l10n_it_edi_related_document/models/account_move.py

```diff
@@ -0,0 +1,113 @@
+from odoo import api, fields, models
+
+from odoo.addons.l10n_it_edi.models.account_move import get_text
+
+
+class AccountMoveRelatedDocumentType(models.Model):
+    _name = "account.move.related_document"
+    _description = "E-invoice Related Documents"
+
+    type = fields.Selection(
+        [
+            ("order", "Order"),
+            ("contract", "Contract"),
+            ("agreement", "Agreement"),
+            ("reception", "Reception"),
+            ("invoice", "Related Invoice"),
+        ],
+        "Document",
+        required=True,
+    )
+    name = fields.Char("Document ID", size=20, required=True)
+    lineRef = fields.Integer("Line Ref.")
+    invoice_id = fields.Many2one(
+        "account.move",
+        "Related Invoice",
+        ondelete="cascade",
+        index=True,
+        readonly=True,
+    )
+    date = fields.Date()
+    numitem = fields.Char("Item Num.", size=20)
+    code = fields.Char("Order Agreement Code", size=100)
+    cig = fields.Char("CIG Code", size=15)
+    cup = fields.Char("CUP Code", size=15)
+    invoice_line_id = fields.Many2one(
+        "account.move.line",
+        "Related Invoice Line",
+        ondelete="cascade",
+        index=True,
+        readonly=True,
+    )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        line_obj = self.env["account.move.line"]
+        for vals in vals_list:
+            if vals.get("lineRef") and not vals.get("invoice_line_id"):
```

Comment:
```suggestion
            if vals.get("lineRef") and not vals.get("invoice_line_id") and vals.get("invoice_id"):
```
Altrimenti quando più sotto fai `vals["invoice_id"]` potrebbe essere sollevato un `KeyError`.

Reply:
:+1: